### PR TITLE
fix: add path alias for src

### LINKED ISSUE #5351: Confusing behavior with tsconfig and alias in some cases
### Describe the bug

If you import something from a folder which isn't aliased, intellisense will suggest imports which are resolvable by TypeScript but not Vite.

### Reproduction

Consider this folder structure:
```
src
  | utils
    | x.ts
  | routes
    | index.svelte
```

If you import something from `routes` that is within `utils` you'll get the following suggestion: `import X from "src/utils/x` . Reason: The generated `tsconfig.json` contains `"baseUrl": "."`. In the eyes of TS this is a correct import path due to the base path alias, in the eyes of Vite it's not because we don't define such an alias (only `$lib` by default).

I'm not sure of the best solution here. Maybe we should always add an alias to Vite that resolves `src/..` to `real/path/to/src`.

### Logs

_No response_

### System Info

```shell
System:                                                    
    OS: Windows 10 10.0.19044                                
    CPU: (12) x64 Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz   
    Memory: 13.04 GB / 31.79 GB                              
  Binaries:                                                  
    Node: 16.15.1 - C:\Program Files\nodejs\node.EXE         
    npm: 8.13.2 - C:\Program Files\nodejs\npm.CMD            
  Browsers:                                                  
    Chrome: 103.0.5060.66                                    
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.44)
    Internet Explorer: 11.0.19041.1566                       
  npmPackages:                                               
    @sveltejs/adapter-node: 1.0.0-next.78 => 1.0.0-next.78   
    @sveltejs/kit: 1.0.0-next.357 => 1.0.0-next.357          
    svelte: 3.48.0 => 3.48.0
```

### Severity

annoyance

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by dummdidumm**
Fixes #5351
Also a drive-by-remove of some obsolete code

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/exports/vite/utils.js [hunk 1]
```diff
@@ -90,6 +91,16 @@ export function get_config_aliases(config) {
 		{ find: '$lib', replacement: config.files.lib }
 	];
 
+	// Our generated tsconfig.json sets a path alias. This causes imports like src/utils to be valid for
+	// TypeScript (its auto completion also suggests it). However, Vite doesn't know about this,
+	// so we do a best-effort attempt to make it work by adding a Vite alias assuming the default
+	// SvelteKit project structure.
+	// For SvelteKit 2.0 we could think about getting rid of the baseUrl which would make this unnecessary
```
Comment (benmccann) [inline packages/kit/src/exports/vite/utils.js:98]: maybe we could write this in a way that would be easier to find via a `grep` when v2 comes around and we want to find all the things to clean up
```suggestion
	// TODO(v2): we could think about getting rid of the baseUrl which would make this unnecessary
```

**Comment by Rich-Harris** [comment]
Eesh, I really don't like this at all. What if we just detected the case where people have added non-relative paths to their own `tsconfig.json` and printed a message telling them they need to fix it? Still _technically_ a breaking change but only one that the most hardcode semver dorks would care about.

**Comment by dummdidumm** [comment]
What specifically don't you like about it? The fact that a silent fallback alias is added, or that it's only working for `src`, or both?

**Comment by benmccann** [comment]
I do think I prefer Rich's solution. It's pretty strange to create an alias from `src` to `src`. We'll have to clean it up later for v2 as it is in this PR, but if we can avoid creating the need to do a later cleanup then that's nicer. I think since this only affects `tsconfig.json` it's not really much of a breaking change

**Comment by dummdidumm** [comment]
"much of a" is still a a little breaking change - I'm not sure we can do this at this point.
The alias from `src` to `src` looks like that due to the test, but in reality it's an alias from `src` to `/path/to/your/project/src` just like `$lib` and all the other custom aliases the user defines.

**Comment by Rich-Harris** [comment]
The part I don't like is that TypeScript will auto-import `src/utils` — that's something we should treat as a bug, not something to bless. I hadn't actually noticed it was happening in my own projects, but if I did my reaction would be 'how the fuck does this work?' and 'what other strange magic is SvelteKit doing?' and 'I don't like this'. We should fix the tsconfig.

The reality of semver is that [every change is a breaking change](https://xkcd.com/1172/), and it's entirely within our remit to apply discretion. Hell, if we auto-updated people's tsconfigs to use relative paths then this actually _wouldn't_ be a breaking change in the sense of 'I upgraded and my dev server no longer starts', though that would probably be a step too far. As long as the blast radius is small (I can't imagine that many people are changing `paths`) and we explain the rationale and the remedy, people will forgive us.

File: .changeset/olive-grapes-lay.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+[fix] add Vite path alias for src
```
Comment (benmccann) [inline .changeset/olive-grapes-lay.md:5]:
```suggestion
fix: add Vite path alias for src
```

**Comment by dummdidumm** [comment]
I adjusted it so that people who set `paths` will have the `baseUrl` set for them. We can remove that check for v2. This feels like the best tradeoff.

**Comment by Rich-Harris** [comment]
Making it non-breaking is a good idea, though I still think we should guide people to the correct outcome, so it now prints a message telling people to use relative imports (and why). The explanation links back to this PR.

**Comment by dummdidumm** [comment]
Changes look good!

**Comment by Rich-Harris** [comment]
(working on the failing unit tests)

File: packages/kit/src/core/sync/write_tsconfig.js [hunk 3]
```diff
@@ -96,43 +125,55 @@ export function write_tsconfig(config, cwd = process.cwd()) {
...
 			const lib = user_paths['$lib'] || [];
 			/** @type {string[]} */
 			const lib_ = user_paths['$lib/*'] || [];
 
+			// TODO(v2): check needs to be adjusted when we remove the base path
 			const missing_lib_paths =
-				!lib.some((relative) => path.resolve(cwd, relative) === config.files.lib) ||
-				!lib_.some((relative) => path.resolve(cwd, relative) === path.join(config.files.lib, '/*'));
+				!lib.some((relative) => path.resolve(cwd, relative) === kit.files.lib) ||
+				!lib_.some((relative) => path.resolve(cwd, relative) === path.join(kit.files.lib, '/*'));
```
Comment (dummdidumm) [inline packages/kit/src/core/sync/write_tsconfig.js:168]: Ah wait, I think we need to adjust this code now already, now that the detection of baseUrl is smarter. It currently advises people to do `src/lib` I think but it should be `./src/lib`

Comment (Rich-Harris) [reply packages/kit/src/core/sync/write_tsconfig.js:168]: done. not sure why the tests are currently failing, they're referencing a bug on line 93 of write_tsconfig.js that can't possibly exist

**Comment by Rich-Harris** [comment]
the fuck is going on with these test failures

**Comment by dummdidumm** [comment]
Thought this might be something with Git, so I created #8568, but my hunch turned out wrong, but I found the reason soon afterwards - therefore closing this in favor of #8568

**Pull request closed without merging**